.travis.yml

```diff
@@ -5,7 +5,8 @@ before_cache:
   - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
   - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
 after_script:
-  - 'if [[ ! -z "$GIT_PASSWORD" ]]; then ./gradlew githubReport; fi'
+  - if [[ ! -z "$GIT_PASSWORD" ]]; then ./gradlew githubReport; fi
+  - if [[ "$TRAVIS_BRANCH" = "master" -a "$TRAVIS_PULL_REQUEST" = "false" ]]; then ./gradlew gitPublishPush; fi
```

Comment:
Same idea here - how about passing the branch to the task: `gitPublishPush "$TRAVIS_BRANCH"`?

Reply:
🤔 that's a good idea. The issues is that `TRAVIS_BRANCH` is the target branch for PRs unless the build is not a PR. I want this to be executed only when we are building master. Should we add the logic to Gradle as we have for Marathon?

```groovy
task master {
  description "Build target for master branch."
}
task pr {
  description "Build target for pull requests."
}
task ci {
  description "Build target for CIs such as Travis CI or Jenkins."
  if (isMasterBuild) dependsOn(master)
  else dependsOn(pr)
}
```

Reply:
Sgtm

Reply:
+1